src/data/vector/vectorFn.ts

```diff
@@ -27,9 +27,9 @@ export const join: (list: PLVector<PLString>) => PLString = (list) => {
   return list.reduce(plString(''), add)
 }
 
-export const joinWith: (separator: PLString, list: PLVector<PLString>) => PLString = (separator, list) => {
+export const joinWith: (separator: PLString, list: PLVector<PLString|PLNumber>) => PLString = (separator, list) => {
```

Comment:
this is wrong, we should not allow implicit type conversation in any PL functions

here if the user pass number it will be implicitly converted into string

Reply:
@maxinteger I've created a separate function for numbers. Can you please take a look?